docs/proposals/innerloop-to-outerloop.md

```diff
@@ -0,0 +1,138 @@
+* Dockerfile, 
+* Buildpacks-v3 and 
+* source-to-image.
+
+### Deploy guidance
+
+In the context of outer-loop development, unless otherwise specified, "Deploy guidance" refers to the guidance a developer or runtime stack author would provide to `odo` for deploying the built image using
+
+* Helm charts
+* Kubernetes manifests, etc.
+
+
+
+## Usage of Build and Deploy guidance in CI/CD and Gitops
+
+
+### Summary 
+
+
+`odo push` syncs user's source code to a workspace container  run using the guidance from a runtime-specific `devfile`.
+
+Example, a Sprintboot project, when associated with a Springboot specific devfile, provides `odo`  the know-how to run the inner-loop container with workspace dependencies
+like debug tools and SDKs.
+
+`odo deploy` would read guidance from the devfile on how to build or more images and deploy the same. The devfile would either be a registry-provided devfile based on the user's choice of runtime or a project-specific devfile that was present at the root of the project.
+
+`odo pipelines` would generate configuration and manifests locally on the filesystem to eventually manage the application component from Git using CI/CD pipelines. The manifests are to be generated based on the build and deploy guidance from the devfile.
+
+### The outerloop and Gitops story
+
+
+In the initial iterations of supporting outerloop, the `odo pipelines` command would be used to generate Tekton and Argo manifests for CI and CD, respectively, for enabling users to drive Gitops-based workflows, organized in an opinionated directory structure in Git.
+
+To enable generation of runtime-specific near-accurate pipelines, `odo` would read the image build and deploy guidance from the devfile, the same way `odo deploy` would 
+do an outerloop build and image deployment on the cluster using guidance from the devfile.
+
+#### Build
+As an example, for a devfile associated with an application component, a dockerfile-based image build guidance in the `devfile` would be consumed by the `odo pipelines` command to
+* generate [Tekton Task yaml manifests](https://github.com/tektoncd/catalog/blob/master/task/buildah/0.1/buildah.yaml) which does a Dockerfile build, or 
+* generate an equivalent [k8s-build manifest](https://github.com/redhat-developer/build/blob/master/samples/build/build_buildah_cr.yaml), orchestrated by Tekton Pipelines.
+* directly use a Tekton Task referenced in the `devfile`.
+
+#### Deploy
+Similarly, the `odo pipelines` command would generate 
+* The necessary 'yaml' manifests to deploy the built image using the guidance from the devfile.
+* The Argo CRs that represent apps that would be reconciled by Argo on a given cluster.
+
+The generated manifests may optionally be consumed outside the context of `odo` using plain old `kubectl` and `kustomize`. Overtime, the user may choose to modify them and commit them
```

Comment:
If the manifests are **optionally** consumed outside of the context of `odo`. Does that mean that there would be a way to consume them in `odo`?

Reply:
`odo` would have the capability to parse the structure and makes changes to it, locally. Example, when someone is adding a new app to the same gitops structure.